### PR TITLE
DI XATTRs / Docker improvements

### LINKED ISSUE #1199: XATTRs: Run accel (di) tests w/ xattrs enabled
Ability to run both sets of di functional tests with xattrs enabled:

 - [ ] accel config(s) should be set to use xattrs, import=continuous
 - [ ] sg config(s) should be set to use xattrs, but not import

### PR DESCRIPTION
**Pull request by sethrosetter**
#### Fixes #1199 .

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- Support running docker in jenkins
- Cleaner api for creating / destroying docker cluster
- Enable running di mode with xattrs

### CODE REVIEW

**Comment by sethrosetter** [comment]
Waiting on (Need close review)
- http://uberjenkins.sc.couchbase.com/job/cen7-sync-gateway-functional-tests-base-di/565/
- http://uberjenkins.sc.couchbase.com/job/cen7-sync-gateway-functional-tests-topology-specific-di/460/

File: docker/cluster.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+from docker.errors import NotFound
+from keywords.exceptions import DockerError
+from keywords.utils import log_info
+
+
+def setup_port_mapping(current_port):
+    """ Expose ports to host for local development """
+
+    port_map = {}
+    ports_to_map = ['8091/tcp', '4984/tcp', '4985/tcp']
```
Comment (raghusarangapani) [inline docker/cluster.py]: Do we care about SSL ports like 18091

File: docker/cluster.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+    return port_map, current_port
+
+
+def create_and_bind_tmp_dirs(container_name, volume_map):
+    """ Create directories for binding the log directories for the various services.
+    This provides quicker inspection / collection of the logs during devlopment """
+
+    tmp_dir_sg = '/tmp/{}-sg'.format(container_name)
+    tmp_dir_cbs = '/tmp/{}-cbs'.format(container_name)
```
Comment (raghusarangapani) [inline docker/cluster.py]: Both sg and cbs have the same container_name?

File: docker/cluster.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+    # Create docker network with name
+    log_info('Creating bridged network: {} ...'.format(network_name))
+    network = docker_client.networks.create(network_name)
+
+    # Loop through nodes, start them on the network that was just created
+    log_info('Starting {} containers on network {} ...'.format(number_of_nodes, network_name))
+    container_names = ['{}.{}'.format(network_name, i) for i in range(number_of_nodes)]
+
+    current_port = 30000
```
Comment (raghusarangapani) [inline docker/cluster.py]: Is this just a random number?

File: docker/mobile-testkit/entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+# Instal repo requirements
+# This may have changed sinces last docker image build
+pip install -r requirements.txt
+
+# Generate cluster configs
+python libraries/utilities/generate_clusters_from_pool.py
+
+# Run tests (--ci)
+pytest --ci $2 --mode=$3 $4 --server-version=$5 --sync-gateway-version=$6 $7 $8
```
Comment (raghusarangapani) [inline docker/mobile-testkit/entrypoint.sh]: Using getopts for parsing command line options might be better. It is similar to python's argparse. That way we need not worry about the order

File: docker/cluster.py (hunk 3 above)
Comment (sethrosetter) [inline docker/cluster.py]: Yes.

File: docker/cluster.py (hunk 2 above)
Comment (sethrosetter) [inline docker/cluster.py]: We are not really sure what will be installed so it sets up bindings for any possible service.

File: docker/cluster.py (hunk 1 above)
Comment (sethrosetter) [inline docker/cluster.py]: Not yet. We can extend support later.

File: docker/mobile-testkit/entrypoint.sh (hunk 4 above)
Comment (sethrosetter) [inline docker/mobile-testkit/entrypoint.sh]: Good call. It is working now. Do we need to block the PR on it or can we file a ticket and do it later?

Comment (raghusarangapani) [inline docker/mobile-testkit/entrypoint.sh]: Later is fine.

**Pull request merged**